moveit_core/collision_distance_field/src/collision_distance_field_types.cpp

```diff
@@ -539,25 +540,26 @@ void collision_detection::getProximityGradientMarkers(
 }
 
 void collision_detection::getCollisionMarkers(
-    const std::string& frame_id, const std::string& ns, const ros::Duration& dur,
+    const std::string& frame_id, const std::string& ns, const rclcpp::Duration& dur,
     const std::vector<PosedBodySphereDecompositionPtr>& posed_decompositions,
     const std::vector<PosedBodySphereDecompositionVectorPtr>& posed_vector_decompositions,
-    const std::vector<GradientInfo>& gradients, visualization_msgs::MarkerArray& arr)
+    const std::vector<GradientInfo>& gradients, visualization_msgs::msg::MarkerArray& arr)
 {
+  rclcpp::Clock ros_clock;
   if (gradients.size() != posed_decompositions.size() + posed_vector_decompositions.size())
   {
-    ROS_WARN_NAMED("collision_distance_field", "Size mismatch between gradients %zu and decompositions %zu",
+    RCLCPP_WARN(LOGGER_COLLISION_DISTANCE_FIELD, "Size mismatch between gradients %zu and decompositions %zu",
                    gradients.size(), posed_decompositions.size() + posed_vector_decompositions.size());
     return;
   }
   for (unsigned int i = 0; i < gradients.size(); i++)
   {
     for (unsigned int j = 0; j < gradients[i].types.size(); j++)
     {
-      visualization_msgs::Marker sphere_mark;
-      sphere_mark.type = visualization_msgs::Marker::SPHERE;
+      visualization_msgs::msg::Marker sphere_mark;
+      sphere_mark.type = visualization_msgs::msg::Marker::SPHERE;
       sphere_mark.header.frame_id = frame_id;
-      sphere_mark.header.stamp = ros::Time::now();
+      sphere_mark.header.stamp = ros_clock.now();
```

Comment:
http://docs.ros2.org/bouncy/api/rclcpp/classrclcpp_1_1_clock.html

This needs to be RCL_ROS_TIME not RCL_SYSTEM_TIME which is the default

Reply:
https://github.com/ros-planning/moveit2/pull/65/commits/65137dba2012aef65d4fd181c21c54f4af2bfed0